budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -1,247 +1,959 @@
-        self.sources = [
-            "Education Infrastructure Grant",
-            "Community Library Service Grant",
-        ]
-        for i in range(30):
-            if i < 15:
-                status = self.statuses[0]
-                province = self.provinces[0]
-                source = self.sources[0]
+        self.quarter = Quarter.objects.create(number=3)
+        self.date = date(2050, 1, 1)
+        self.irm_snapshot = IRMSnapshot.objects.create(
+            financial_year=self.fin_year, quarter=self.quarter, date_taken=self.date
+        )
+        self.project = ProvInfraProject.objects.create(IRM_project_id=123456)
+        ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot,
+            project=self.project,
+            name="Project 123456",
+            department="Health",
+            province="Western Cape",
+            status="Construction",
+            primary_funding_source="Health Infrastructure Grant",
+            estimated_completion_date=date(year=2020, month=1, day=1)
+        )
+        # Add ten projects
+        provinces = ["Eastern Cape", "Free State"]
+        for i in range(10):
+            if i < 5:
+                province = provinces[0]
             else:
-                status = self.statuses[1]
-                province = self.provinces[1]
-                source = self.sources[1]
-
-            ProvInfraProject.objects.create(
-                financial_year=self.fin_year,
-                IRM_project_id=i,
+                province = provinces[1]
+            project = ProvInfraProject.objects.create(IRM_project_id=i)
+            ProvInfraProjectSnapshot.objects.create(
+                irm_snapshot=self.irm_snapshot,
+                project=project,
                 name="Project {}".format(i),
-                department="Department {}".format(i),
-                local_municipality="Local {}".format(i),
-                district_municipality="District {}".format(i),
                 province=province,
-                status=status,
-                primary_funding_source=source,
-                main_contractor="Contractor {}".format(i),
-                principle_agent="Principle Agent {}".format(i),
-                program_implementing_agent="Program Agent {}".format(i),
-                other_parties="Service Provider: XXXX{}".format(i),
+                estimated_completion_date=date(year=2020, month=1, day=1)
             )
+        ProvInfraProjectIndex().reindex()
 
-    def test_projects_per_page(self):
-        response = self.client.get(self.url, format="json")
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    def tearDown(self):
+        ProvInfraProjectIndex().clear()
+
+    def test_correct_numbers_showed(self):
+        selenium = self.selenium
+        selenium.get("%s%s" % (self.live_server_url, self.url))
+        num_of_projects = selenium.find_element_by_xpath(
```

Comment:
use `find_element_by_css_selector` rather than `find_element_by_xpath` - it's a lot cleaner and more like what's used in web programming

Reply:
all find_element_by_xpath methods changed with find_element_by_css_selector